epochStart/shardchain/trigger.go

```diff
@@ -735,6 +744,15 @@ func (t *trigger) getHeaderWithNonceAndPrevHash(nonce uint64, prevHash []byte) (
 	return t.getHeaderWithNonceAndHash(nonce, b.Data[0])
 }
 
+func (t *trigger) getAllFinishedStartOfEpochMetaHdrs() []*block.MetaBlock {
+	finishedMetaHdrs := make([]*block.MetaBlock, 0, len(t.mapFinalizedEpochs))
```

Comment:
You can define this var like this: finishedMetaHdrs := make([]*block.MetaBlock, len(t.mapFinalizedEpochs)) and use 
finishedMetaHdrs[index] instead append in the next for

Reply:
no, as the index is efectively equal with epoch. so you should have a separate index counter.

Reply:
ok